robot/client/client.go

```diff
@@ -991,10 +991,27 @@ func (rc *RobotClient) RestartModule(ctx context.Context, req robot.RestartModul
 	return nil
 }
 
-// Shutdown shuts down the robot. Will return "shutdown function not defined" error if
-// shutdown callback is not specified in robot options.
+// Shutdown shuts down the robot. May return DeadlineExceeded error if shutdown request times out,
+// or if robot server shuts down before having a chance to send a response.
 func (rc *RobotClient) Shutdown(ctx context.Context) error {
 	reqPb := &pb.ShutdownRequest{}
 	_, err := rc.client.Shutdown(ctx, reqPb)
-	return err
+	if err != nil {
+		status, ok := status.FromError(err)
+		if ok {
+			switch status.Code().String() {
+			case "Unavailable", "Internal", "Unknown":
```

Comment:
Are there actual gRPC constants for these? As opposed to just strings?

Reply:
yes there are but the linter complains that the switch isn't exhaustive, so I thought it was cleaner to just check the strings than have another case for every other error code

Reply:
would prefer just adding cases for the other constants